op-acceptance-tests/tests/batcher/batcher_test.go

```diff
@@ -0,0 +1,130 @@
+
+	latestUnsafe_A := sys.L2CL.StopSequencer()
+	l.Info("Latest unsafe block after stopping the L2 sequencer", "latestUnsafe", latestUnsafe_A)
+
+	parent := latestUnsafe_A
+	for j := 0; j < 200; j++ {
+		l1Origin := sys.L1EL.BlockRefByLabel(eth.Unsafe).Hash
+
+		for i := 0; i < 5; i++ {
+			l.Debug("Sequencing L2 block", "iteration", i, "parent", parent)
+			sequenceBlockWithL1Origin(t, ts_L2, parent, l1Origin, alice, cathrine)
+
+			parent = sys.L2CL.HeadBlockRef(types.LocalUnsafe).Hash
+
+			sys.AdvanceTime(time.Second * 2)
+		}
+
+		l.Debug("Sequencing L1 block", "iteration_j", j)
+		sequenceBlock(t, ts_L1, common.Hash{})
+	}
+
+	sys.L2CL.StartSequencer()
+
+	l.Info("Current L1 unsafe block", "currentL1Unsafe", sys.L1EL.BlockRefByLabel(eth.Unsafe))
+	sys.ControlPlane.FakePoSState(cl.ID(), stack.Start)
+
+	sys.L2Batcher.Start()
+
+	channels, channelFrames, l2Txs := sys.L2Chain.DeriveData(4) // over the next 4 blocks, collect batches/channels/frames submitted by the batcher on the L1 network, and parse them
+	{
+		for _, c := range channels {
+			l.Info("Channel details", "channelID", c.String(), "frameCount", len(channelFrames[c]), "dataLength_frame0", len(channelFrames[c][0].Data))
+		}
+
+		require.Equal(t, 10, len(channels)) // we expect a total of 10 channels, due to existing MaxChannelDuration bug filed at: https://github.com/ethereum-optimism/optimism/issues/18092
+
+		// values are dependent on:
+		// - MaxPendingTransactions
+		// - number of blocks and transactions sent in the test - 1000 L2 blocks with 1 transaction from cathrine to alice
+		// - MaxL1TxSize (this is set to 40_000 bytes for test purposes)
+		sizeRanges := []struct {
+			min  int
+			max  int
+			note string
+		}{
+			{min: 7_000, max: 10_000, note: "channel 0 - the first 100 blocks"},
+			{min: 100, max: 1000, note: "channel 1 - only a few blocks due to racy behavior, nowhere near the channel capacity"},
+			{min: 100, max: 1000, note: "channel 2 - only a few blocks due to racy behavior, nowhere near the channel capacity"},
+			{min: 100, max: 1000, note: "channel 3 - only a few blocks due to racy behavior, nowhere near the channel capacity"},
+			{min: 100, max: 1000, note: "channel 4 - only a few blocks due to racy behavior, nowhere near the channel capacity"},
+			{min: 100, max: 1000, note: "channel 5 - only a few blocks due to racy behavior, nowhere near the channel capacity"},
+			{min: 100, max: 1000, note: "channel 6 - only a few blocks due to racy behavior, nowhere near the channel capacity"},
+			{min: 100, max: 1000, note: "channel 7 - only a few blocks due to racy behavior, nowhere near the channel capacity"},
+			{min: 20_000, max: 40_000, note: "channel 8 - filled to the max capacity, due to blocking behavior because of MaxPendingTransactions limit reached"},
+			{min: 20_000, max: 40_000, note: "channel 9 - filled to the max capacity, due to blocking behavior because of MaxPendingTransactions limit reached"},
+		}
```

Comment:
Confirmed that that's the case on a devnet too.

<img width="368" height="157" alt="Image" src="https://github.com/user-attachments/assets/23b24b3f-5040-417f-8ff0-a349dafdffe1" />
<img width="2159" height="1126" alt="Image" src="https://github.com/user-attachments/assets/6fbf342c-1ff5-489e-a86f-4666bb6d7533" />